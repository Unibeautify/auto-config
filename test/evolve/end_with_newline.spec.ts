import { testGenetic } from "../testGenetic";
import prettier from "@unibeautify/beautifier-prettier";
import jsBeautify from "@unibeautify/beautifier-js-beautify";
import fileBeautifier from "@unibeautify/beautifier-file";

testGenetic("end_with_newline1", {
  beautifiers: [prettier, jsBeautify, fileBeautifier],
  language: "JavaScript",
  originalText: `if (true) { helloWorld();\n helloWorld();\n}`,
  desiredText: `if (true) {\r\nhelloWorld();\r\nhelloWorld();\r\n}\r\n`,
});

testGenetic("end_with_newline2", {
  beautifiers: [prettier, jsBeautify, fileBeautifier],
  language: "JavaScript",
  originalText: `if (true) { helloWorld();\n helloWorld();\n}`,
  desiredText: `if (true) {\nhelloWorld();\nhelloWorld();\n}\n`,
});

testGenetic("end_with_newline3", {
  beautifiers: [prettier, jsBeautify, fileBeautifier],
  language: "JavaScript",
  originalText: `if (true) { helloWorld();\n helloWorld();\n}`,
  desiredText: `if (true) {\nhelloWorld();\nhelloWorld();\n}`,
});
