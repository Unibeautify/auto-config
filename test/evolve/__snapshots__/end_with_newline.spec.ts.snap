// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end_with_newline1: beautifiedText 1`] = `
if (true) {
helloWorld();
helloWorld();
}

`;

exports[`end_with_newline1: beautifiers 1`] = `
Array [
  "Prettier",
  "File",
]
`;

exports[`end_with_newline1: trimmedOptions 1`] = `
Object {
  "end_of_line": "CRLF",
  "indent_size": 0,
}
`;

exports[`end_with_newline2: beautifiedText 1`] = `
if (true) {
helloWorld();
helloWorld();
}

`;

exports[`end_with_newline2: beautifiers 1`] = `
Array [
  "Prettier",
]
`;

exports[`end_with_newline2: trimmedOptions 1`] = `
Object {
  "indent_size": 0,
}
`;

exports[`end_with_newline3: beautifiedText 1`] = `
if (true) {
helloWorld();
helloWorld();
}
`;

exports[`end_with_newline3: beautifiers 1`] = `
Array [
  "Prettier",
  "File",
]
`;

exports[`end_with_newline3: trimmedOptions 1`] = `
Object {
  "end_with_newline": false,
  "indent_size": 0,
}
`;
